source/simulator/helper_functions.cc

```diff
@@ -2275,13 +2275,28 @@ namespace aspect
   {
     const Quadrature<dim-1> &quadrature_formula = introspection.face_quadratures.temperature;
 
+    bool consider_darcy_velocity = false;
+    unsigned int porosity_idx = numbers::invalid_unsigned_int;
+    if (introspection.composition_type_exists(CompositionalFieldDescription::porosity))
+      {
+        porosity_idx = introspection.find_composition_type(CompositionalFieldDescription::porosity);
+        if (introspection.compositional_field_methods[porosity_idx] == Parameters<dim>::AdvectionFieldMethod::fem_darcy_field)
+          consider_darcy_velocity = true;
+      }
+
     FEFaceValues<dim> fe_face_values (*mapping,
                                       finite_element,
                                       quadrature_formula,
                                       update_values   | update_normal_vectors |
                                       update_quadrature_points | update_JxW_values);
 
     std::vector<Tensor<1,dim>> face_current_velocity_values (fe_face_values.n_quadrature_points);
+    std::vector<Tensor<1,dim>> face_current_fluid_velocity_values(fe_face_values.n_quadrature_points);
+
+    MaterialModel::MaterialModelInputs<dim> in(fe_face_values.n_quadrature_points, introspection.n_compositional_fields);
+    MaterialModel::MaterialModelOutputs<dim> out(fe_face_values.n_quadrature_points, introspection.n_compositional_fields);
+    MeltHandler<dim>::create_material_model_outputs(out);
+    MaterialModel::MeltOutputs<dim> *fluid_out = nullptr;
```

Comment:
I don't see where this is being initialized but you are reading from this below.